pkg/logger/logger.go

```diff
@@ -13,115 +15,103 @@ const (
+		l.SetWriter(newLogger.writer)
+	}
 }
 
 // Debug prints a message when Debugg is set to true on the Logger
 func (l Logger) Debug(format string, a ...interface{}) {
 	if l.Debugg {
 		message := fmt.Sprintf(format, a...)
-		Println(message)
+		l.println(message)
 	}
 }
 
 // Verbose prints a message when Verbosee is set to true on the Logger
 func (l Logger) Verbose(format string, a ...interface{}) {
 	if l.Verbosee {
 		message := fmt.Sprintf(format, a...)
-		Println(message)
+		l.println(message)
 	}
 }
 
 // Warning prints a warning message to Stdout in yellow
 func (l Logger) Warning(format string, a ...interface{}) {
 	message := fmt.Sprintf(format, a...)
 	if l.NoColor {
-		Println(message)
+		l.println(message)
 	} else {
-		PrintlnColor(message, YELLOW)
-	}
-}
-
-func (l Logger) PrintLogMessage(message LogMessage) {
-	switch message.Level {
-	case "error":
-		l.Error(message.Message)
-	case "warning":
-		l.Warning(message.Message)
-	case "debug":
-		l.Debug(message.Message)
-	case "verbose":
-		l.Verbose(message.Message)
-	default:
-		l.Output(message.Message)
+		l.printlnColor(message, YELLOW)
 	}
 }
 
-func (l Logger) PrintLogMessages(messages []LogMessage) {
-	for _, message := range messages {
-		l.PrintLogMessage(message)
-	}
-}
-
-// Warning prints a warning message to Stdout in yellow
-func Warning(format string, a ...interface{}) {
-	message := fmt.Sprintf(format, a...)
-	PrintlnColor(message, YELLOW)
-}
-
 // Output prints a message on Stdout in 'normal' color
 func (l Logger) Output(format string, a ...interface{}) {
 	message := fmt.Sprintf(format, a...)
-	Println(message)
-}
-
-// Output prints an error message to Stdout in 'normal' color
-func Output(format string, a ...interface{}) {
-	message := fmt.Sprintf(format, a...)
-	Println(message)
+	l.println(message)
 }
 
 // Error prints an error message to Stdout in red
 func (l Logger) Error(format string, a ...interface{}) {
 	message := fmt.Sprintf(format, a...)
 	if l.NoColor {
-		Println(message)
+		l.println(message)
 	} else {
-		PrintlnColor(message, RED)
+		l.printlnColor(message, RED)
 	}
 }
 
-// Error prints an error message to Stdout in red
-func Error(format string, a ...interface{}) {
-	message := fmt.Sprintf(format, a...)
-	PrintlnColor(message, RED)
-}
-
-// Print prints a message
-func Print(message string) {
-	fmt.Printf("%s", message)
-}
-
-// Println prints a message with a trailing newline
-func Println(message string) {
-	fmt.Printf("%s\n", message)
+// println prints a message with a trailing newline
+func (l Logger) println(message string) {
+	l.LazyInit()
+	fmt.Fprintf(l.writer, "%s\n", message)
 }
 
-// PrintColor prints a message in a given ANSI-color
-func PrintColor(message string, color string) {
-	fmt.Printf("%s%s%s", color, message, RESET)
+// printColor prints a message in a given ANSI-color
+func (l Logger) printColor(message string, color string) {
+	l.LazyInit()
+	fmt.Fprintf(l.writer, "%s%s%s", color, message, RESET)
```

Comment:
RESET variable looks abnormal (its case)

First a full uppercase is wrongly discouraged in Go

Then it also means it's exported.

Do we really need this?

But this remark is out of PR scope as RED, YELLOW, RESET … were present before this current PR

Reply:
I renamed them by normalizing their names and prefixing them with `escSeq` which should help people unfamiliar with escape sequences to recognize them as such.